go/vt/wrangler/reparent.go

```diff
@@ -876,110 +883,102 @@ func (wr *Wrangler) emergencyReparentShardLocked(ctx context.Context, ev *events
 		return err
 	}
 
-	// Check invariants we're going to depend on.
-	masterElectTabletAliasStr := topoproto.TabletAliasString(masterElectTabletAlias)
-	masterElectTabletInfo, ok := tabletMap[masterElectTabletAliasStr]
-	if !ok {
-		return fmt.Errorf("master-elect tablet %v is not in the shard", masterElectTabletAliasStr)
-	}
-	ev.NewMaster = *masterElectTabletInfo.Tablet
-	if topoproto.TabletAliasEqual(shardInfo.MasterAlias, masterElectTabletAlias) {
-		return fmt.Errorf("master-elect tablet %v is already the master", topoproto.TabletAliasString(masterElectTabletAlias))
-	}
-
-	// Deal with the old master: try to remote-scrap it, if it's
-	// truly dead we force-scrap it. Remove it from our map in any case.
-	if shardInfo.HasMaster() {
-		deleteOldMaster := true
-		shardInfoMasterAliasStr := topoproto.TabletAliasString(shardInfo.MasterAlias)
-		oldMasterTabletInfo, ok := tabletMap[shardInfoMasterAliasStr]
-		if ok {
-			delete(tabletMap, shardInfoMasterAliasStr)
-		} else {
-			oldMasterTabletInfo, err = wr.ts.GetTablet(ctx, shardInfo.MasterAlias)
-			if err != nil {
-				wr.logger.Warningf("cannot read old master tablet %v, won't touch it: %v", shardInfoMasterAliasStr, err)
-				deleteOldMaster = false
-			}
-		}
-
-		if deleteOldMaster {
-			ev.OldMaster = *oldMasterTabletInfo.Tablet
-			wr.logger.Infof("deleting old master %v", shardInfoMasterAliasStr)
-
-			ctx, cancel := context.WithTimeout(ctx, waitReplicasTimeout)
-			defer cancel()
-
-			if err := topotools.DeleteTablet(ctx, wr.ts, oldMasterTabletInfo.Tablet); err != nil {
-				wr.logger.Warningf("failed to delete old master tablet %v: %v", shardInfoMasterAliasStr, err)
-			}
-		}
+	statusMap, masterStatusMap, err := wr.stopReplicationAndBuildStatusMaps(ctx, ev, tabletMap, waitReplicasTimeout, unreachableReplicas)
+	if err != nil {
+		return err
 	}
 
-	// Stop replication on all replicas, get their current
-	// replication position
-	event.DispatchUpdate(ev, "stop replication on all replicas")
-	wg := sync.WaitGroup{}
-	mu := sync.Mutex{}
-	statusMap := make(map[string]*replicationdatapb.StopReplicationStatus)
-	for alias, tabletInfo := range tabletMap {
-		wg.Add(1)
-		go func(alias string, tabletInfo *topo.TabletInfo) {
-			defer wg.Done()
-			wr.logger.Infof("getting replication position from %v", alias)
-			ctx, cancel := context.WithTimeout(ctx, waitReplicasTimeout)
-			defer cancel()
-			// TODO: Once we refactor EmergencyReparent, change the stopReplicationOption argument to IOThreadOnly.
-			_, stopReplicationStatus, err := wr.tmc.StopReplicationAndGetStatus(ctx, tabletInfo.Tablet, replicationdatapb.StopReplicationMode_IOANDSQLTHREAD)
-			if err != nil {
-				wr.logger.Warningf("failed to get replication status from %v, ignoring tablet: %v", alias, err)
-				return
-			}
-			mu.Lock()
-			statusMap[alias] = stopReplicationStatus
-			mu.Unlock()
-		}(alias, tabletInfo)
+	if len(statusMap)+len(masterStatusMap) < len(tabletMap)-unreachableReplicas.Len()-1 {
```

Comment:
I initially thought this could potentially happen if contexts time out, but I realized that should cause error returns that would cause us to hit the > 1 error condition check inside of `stopReplicationAndBuildStatusMaps`, so I'll remove this additional check.

Reply:
Done.